src/sider/oversikt/sokeresultat/toolbar/TildelOppfolgingsenhet/hooks/usePostTildelOppfolgingsenhet.tsx

```diff
@@ -0,0 +1,54 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { SYFOBEHANDLENDEENHET_ROOT } from '@/apiConstants';
+import { post } from '@/api/axios';
+import { PersonOversiktStatusDTO } from '@/api/types/personoversiktTypes';
+import { personoversiktQueryKeys } from '@/data/personoversiktHooks';
+import { useAktivEnhet } from '@/context/aktivEnhet/AktivEnhetContext';
+
+export function usePostTildelOppfolgingsenhet() {
+  const queryClient = useQueryClient();
+  const { aktivEnhet } = useAktivEnhet();
+
+  const path = `${SYFOBEHANDLENDEENHET_ROOT}/oppfolgingsenhet-tildelinger`;
+  const postTildelOppfolgingsenhet = (
+    person: OppfolgingsenhetTildelingerRequestDTO
+  ) => post<OppfolgingsenhetTildelingerResponseDTO>(path, person);
+
+  return useMutation({
+    mutationFn: postTildelOppfolgingsenhet,
+    onSuccess: (data: OppfolgingsenhetTildelingerResponseDTO) => {
+      const personer: PersonOversiktStatusDTO[] =
+        queryClient.getQueryData(
+          personoversiktQueryKeys.personoversiktEnhet(aktivEnhet)
+        ) || [];
+
+      const updatedPersoner: PersonOversiktStatusDTO[] = personer.filter(
+        (person) => {
+          const isPersonTildelt = data.tildelinger.find(
+            (tildeling) => tildeling.personident === person.fnr
+          );
+          return !isPersonTildelt;
+        }
+      );
+
+      queryClient.setQueryData(
+        personoversiktQueryKeys.personoversiktEnhet(aktivEnhet),
+        updatedPersoner
+      );
```

Comment:
Dette er for å fjerne de fra oversikten med en gang? Jeg ville tenkt at man bare invaliderte `personoversiktQueryKeys.personoversiktEnhet(aktivEnhet)`, men det er jo ikke sikkert det går instantly her 🤔
Det "rare" her blir vel hvis vi gjør dette, og så refresher de, og da er folka tilbake igjen fordi det tar litt tid. Men vet ikke om det faktisk kommer til å skje.

Reply:
Stemmer! Det er billigere å fikse på cachen enn å hente hele oversikten på ny 😊 Heller ikke sikkert at oversikten er oppdatert i samme sekund som denne returnerer. Hvis de refresher etter 1-2 sek ville jeg tenkt at oversikten var oppdatert. Vi konsumerer vel eventer som dette med en gang 🤔 Hvis ikke burde vi gjøre det